### ISSUE TITLE
Gdx EverythingDemo - ArrayIndexOutOfBoundsException when spinning mous around player

### ISSUE THREAD

**Issue by dasmau89** [SquidPony/SquidLib#67]
I get an exception with the following stacktrace in the EverythingDemo when I spin the mouse around the player for a while:

Exception in thread "LWJGL Application" java.lang.ArrayIndexOutOfBoundsException: 30
	at squidpony.gdx.examples.EverythingDemo$2.mouseMoved(EverythingDemo.java:257)
	at squidpony.squidgrid.gui.gdx.SquidMouse.mouseMoved(SquidMouse.java:183)
	at squidpony.squidgrid.gui.gdx.SquidInput.mouseMoved(SquidInput.java:314)
	at com.badlogic.gdx.backends.lwjgl.LwjglInput.processEvents(LwjglInput.java:312)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication.mainLoop(LwjglApplication.java:207)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:120)

It seems to be easier reproduceable when I move the player to an open area so the fov is bigger.

I noticed this after I tried to figure out what went wrong with the mouse highlight function. The selected path isn't under my mouse cursor in the majority of the time(when I make the window bigger and move the mouse more to the right it becomes more clear, so its aparently a problem with the unprojection from screen coordinates to tile coordinates).
I will create a new issue for this specific part when I figure out how to create a screenshot with the mouse cursor in it on my system, or maybe I get around to have a look at it and create a pull request

**Comment by tommyettinger** [comment]
I just committed a fix for this. It turned out to be a surprisingly obvious mistake in retrospect; I was clamping the values that would be passed to SquidMouse's InputAdapter, but accidentally including the upper bound (here, 30, which is too large for the array). Thanks for spotting this; I'm going to need to ramp up my time spent on bugfixing if it turns out I've made too many mistakes like this, heh. I'm closing this but I still need to get a few more things cleaned up before the 2.9.1 bugfix release.
